examples/ner_new/ner_model_processor.py

```diff
@@ -0,0 +1,277 @@
+        tgt_tokens = tgt_sentence["Token"]
+
+        tgt_ptr, pred_ptr = 0, 0
+
+        for i in range(len(tgt_tokens["text"])):
+            w = tgt_tokens["text"][i]
+            p = tgt_tokens["pos"][i]
+            ch = tgt_tokens["chunk"][i]
+            # TODO: This is not correct and probably we need a utility to do
+            #       BIO encoding to get ner_type?
+            if tgt_ptr < len(tgt_entity_mention["span"]) and \
+                    (tgt_entity_mention["span"][tgt_ptr] ==
+                     tgt_tokens["span"][i]).all():
+                tgt = tgt_entity_mention["ner_type"][tgt_ptr]
+                tgt_ptr += 1
+            else:
+                tgt = "O"
+
+            if pred_ptr < len(pred_entity_mention["span"]) and \
+                    (pred_entity_mention["span"][pred_ptr] ==
+                     tgt_tokens["span"][i]).all():
+                pred = pred_entity_mention["ner_type"][pred_ptr]
+                pred_ptr += 1
+            else:
+                pred = "O"
+
+            opened_file.write(
+                "%d %s %s %s %s %s\n" % (i + 1, w, p, ch, tgt, pred)
+            )
+
+        opened_file.write("\n")
+    opened_file.close()
+
+
+class NerModelProcessor(ModelProcessor):
+    def __init__(self):
+        super().__init__()
+
+        # Load user-defined configuration
+        self.config_model = \
+            yaml.safe_load(open("configs/config_model.yml", "r"))
+        self.config_preprocess = \
+            yaml.safe_load(open("configs/config_preprocessor.yml", "r"))
+
+        self.config = Config({}, default_hparams=None)
+        self.config.add_hparam('config_model', self.config_model)
+        self.config.add_hparam('preprocessor', self.config_preprocess)
+
+        self.model: Optional[nn.Module] = None
+        self.optim: Optional[Optimizer] = None
+
+        self.train_err: float = 0.0
+        self.train_total: float = 0.0
+        self.val_scores: List = []
+        self.output_file = "tmp_eval.txt"
+        self.score_file = "tmp_eval.score"
+        self.scores: Dict[str, float] = {}
+
+        self.logger = logging.getLogger(__name__)
+
+    def _create_model(self,
+                      schemes: Dict[str, Dict[str, BaseExtractor]],
+                      pl_config: Config):
+        text_extractor: BaseExtractor = schemes["text_tag"]["extractor"]
+        char_extractor: BaseExtractor = schemes["char_tag"]["extractor"]
+        ner_extractor: BaseExtractor = schemes["ner_tag"]["extractor"]
+
+        # embedding_dict = \
+        #     load_glove_embedding(config.preprocessor.embedding_path)
+        #
+        # for word in embedding_dict:
+        #     if not text_extractor.contains(word):
+        #         text_extractor.add_entry(word)
+        #
+
+        # TODO: temporarily make fake pretrained emb for debugging
+        embedding_dict = {}
+        fake_tensor = torch.tensor([0.0 for i in range(100)])
```

Comment:
Now you can use the real emb?

Reply:
This file is actually deprecated. Sorry for this. I have the similar logic in another PR and I will make sure use the real emb